rllib/evaluation/collectors/simple_list_collector.py

```diff
@@ -442,17 +452,22 @@ def get_inference_input_dict(self, policy_id: PolicyID) -> \
             # Create the batch of data from the different buffers.
             data_col = view_req.data_col or view_col
             time_indices = \
-                view_req.data_rel_pos - (
+                view_req.shift - (
                     1 if data_col in [SampleBatch.OBS, "t", "env_id",
-                                      SampleBatch.EPS_ID,
                                       SampleBatch.AGENT_INDEX] else 0)
             data_list = []
             for k in keys:
-                if data_col not in buffers[k]:
-                    self.agent_collectors[k]._build_buffers({
-                        data_col: view_req.space.sample()
-                    })
-                data_list.append(buffers[k][data_col][time_indices])
+                if data_col == SampleBatch.EPS_ID:
```

Comment:
same as above: episode_id is always the same within one agent's collector. No need to collect an extra buffer here.